src/dfilemaker/dfilemaker.c

```diff
@@ -564,12 +605,16 @@ int main(int narg, char **arg)
      * initialize mfu and MPI 
      *--------------------------*/
     MPI_Init(&narg, &arg);
+    DTCMP_Init();
     mfu_init();
     mfu_debug_level = MFU_LOG_VERBOSE;
 
     int rank, nrank;
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
     MPI_Comm_size(MPI_COMM_WORLD, &nrank);
+    MPI_Datatype dirname_type;
```

Comment:
@levatin1 , dnamlen has to be defined when we use it to create the MPI datatype.  We'll want to move these three lines down to around line 782 after we assign a value to dnamlen.